contrib/prometheus/recording-rules.yml

```diff
@@ -0,0 +1,5 @@
+groups:
+  - name: host-metering
+    rules:
+    - record: host:usage:workload:cpu_hours1h
+      expr:  max by(_id) (sum_over_time(system_cpu_logical_count[1h:10m])) / scalar(count_over_time(vector(1)[1h:10m]))
```

Comment:
Don't you want to count over time only on samples for the `system_cpu_logical_count` series? 🤔

Reply:
The idea is to count it as relative to the amount of samples. For example, let's assume the following samples:

```
00:10 - 8
00:20 - 8
00:30 - 8
01:10 - 8
```

Now we would like to count it this way for the interval 00:00 - 01:00:
```
8+8+8 / 1+1+1+1+1+1 = 24/6 = 4 cpu/hours
```
The idea is to keep the amount of 10 minute intervals constant.

Reply:
So the system reported 8 cores of CPU from 00:10 to 1:10 and somehow this means 4 cpu/hours? Shouldn't it be 8 cpu/hours? 🤔

Reply:
Yes, because the system did not report for half an hour. We can't assume that it was on during that period of time, hence we will not report the full amount of cpu/hours.

Reply:
Makes sense now. Thanks for the explanation, @ShimShtein